vllm/v1/attention/backends/flashinfer.py

```diff
@@ -92,6 +95,40 @@ def get_kv_cache_stride_order() -> tuple[int, ...]:
             raise ValueError(f"Unknown cache layout format {cache_layout}.")
         return stride_order
 
+    @staticmethod
+    def use_trtllm_decode_attention(batch_size: int,
+                                    max_seq_len: int,
+                                    kv_cache_dtype: str = "auto") -> bool:
+        # Check if environment variable is explicitly set
+        if FlashInferBackend.cached_use_trtllm is not None:
+            return FlashInferBackend.cached_use_trtllm
+        env_value = envs.VLLM_USE_TRTLLM_DECODE_ATTENTION
+        if env_value is not None:
+            logger.info_once("VLLM_USE_TRTLLM_DECODE_ATTENTION is set to %s",
+                             env_value)
+            # Environment variable is set - respect it
+            # Making the conditional check for zero because
+            # the path is automatically enabled if the batch size condition
+            # is satisfied.
+            no_use_trtllm = env_value == "0"
+            if not no_use_trtllm:
+                logger.info_once(
+                    "VLLM_USE_TRTLLM_DECODE_ATTENTION is set to 1, "
+                    "using TRTLLM decode attention.")
+            FlashInferBackend.cached_use_trtllm = not no_use_trtllm
+        else:
+            # Environment variable not set - use auto-detection
+            use_trtllm = (is_sm100a_supported(torch.device("cuda"))
+                          and batch_size <= 256 and max_seq_len < 131072
+                          and kv_cache_dtype == "auto")
+            FlashInferBackend.cached_use_trtllm = use_trtllm
+            if use_trtllm:
+                logger.warning_once(
+                    "Using TRTLLM decode attention (auto-detected)."
+                    "Please run with VLLM_KV_CACHE_LAYOUT=HND for "
+                    "accurate results.")
```

Comment:
I think we might need to make it opt-in if we need to specify VLLM_KV_CACHE_LAYOUT=HND for correctness.. Can we set this automatically?